libs/src/WebAudiusLibs.ts

```diff
@@ -0,0 +1,605 @@
+
+    /** Identity Service */
+    if (this.identityServiceConfig) {
+      this.identityService = new IdentityService({
+        identityServiceEndpoint: this.identityServiceConfig.url
+      })
+      const hedgehogService = new Hedgehog({
+        identityService: this.identityService,
+        useLocalStorage: this.identityServiceConfig.useHedgehogLocalStorage,
+        localStorage: this.localStorage,
+        ...this.hedgehogConfig
+      })
+      this.hedgehog = hedgehogService.instance
+      await this.hedgehog.waitUntilReady()
+    } else if (this.web3Config && !this.web3Config.useExternalWeb3) {
+      throw new Error('Identity Service required for internal Web3')
+    }
+
+    /** Web3 Managers */
+    if (this.ethWeb3Config) {
+      this.ethWeb3Manager = new EthWeb3Manager({
+        web3Config: this.ethWeb3Config,
+        identityService: this.identityService,
+        hedgehog: this.hedgehog
+      })
+    }
+    if (this.web3Config) {
+      this.web3Manager = new Web3Manager({
+        web3Config: this.web3Config,
+        identityService: this.identityService,
+        hedgehog: this.hedgehog,
+        isServer: this.isServer
+      })
+      await this.web3Manager.init()
+      if (this.identityService) {
+        this.identityService.setWeb3Manager(this.web3Manager)
+      }
+    }
+    if (this.solanaWeb3Config) {
+      this.solanaWeb3Manager = new SolanaWeb3Manager(
+        this.solanaWeb3Config,
+        this.identityService,
+        this.web3Manager
+      )
+      await this.solanaWeb3Manager.init()
+    }
+
+    /** Contracts - Eth and Data Contracts */
+    const contractsToInit = []
+    if (this.ethWeb3Manager) {
+      const {
+        tokenAddress = null,
+        registryAddress = null,
+        claimDistributionContractAddress = null,
+        wormholeContractAddress = null
+      } = this.ethWeb3Config ?? {}
+
+      this.ethContracts = new EthContracts({
+        ethWeb3Manager: this.ethWeb3Manager,
+        tokenContractAddress: tokenAddress!,
+        registryAddress: registryAddress!,
+        claimDistributionContractAddress: claimDistributionContractAddress!,
+        wormholeContractAddress: wormholeContractAddress!,
+        isServer: this.isServer,
+        logger: this.logger,
+        isDebug: this.isDebug
+      })
+
+      contractsToInit.push(this.ethContracts.init())
+    }
+    if (this.web3Manager) {
+      this.contracts = new AudiusContracts(
+        this.web3Manager,
+        (this.web3Config ? this.web3Config.registryAddress : null) as string,
+        (this.web3Config
+          ? this.web3Config.entityManagerAddress
+          : null) as string,
+        this.isServer,
+        this.logger
+      )
+      contractsToInit.push(this.contracts.init())
+    }
+    await Promise.all(contractsToInit)
+    if (
+      this.wormholeConfig &&
+      this.ethWeb3Manager &&
+      this.ethContracts &&
+      this.solanaWeb3Manager
+    ) {
+      this.wormholeClient = new ProxyWormhole(
```

Comment:
Yeah let's def test this before merging

Reply:
on client side, since it's a separate entry-point now, im thinking we add a feature flag and then pick on or the other basically.